IHP/FrameworkConfig.hs

```diff
@@ -18,18 +17,26 @@ import IHP.Mail.Types
 defaultPort :: Int
 defaultPort = 8000
 
-portRef :: IORef Int
-portRef = unsafePerformIO (newIORef defaultPort)
+portRef :: IO (IORef Int)
```

Comment:
I think we could actually now get completly rid of this `portRef` by merging `initAppPort` into `defaultFrameworkConfig` :)

Reply:
How do we deal with the portRef write in startToolServer in `IHP/IHP/IDE/ToolServer.hs`? Can it just be removed?

Reply:
It does that to set the port at which the tool server runs. Usually port 8001, but when port 8001 is busy it decides to user other free ports. The tool server is also built with IHP. The `startToolServer'` is similiar to the `IHP.Server` module. Previously the only way for getting this to work with the FrameworkConfig type class was by using an IORef. Now we can just build a right FrameworkConfig value where we just set the port.

Reply:
Best way to test that this still works is by having two different ihp projects and running them at the same time. This should then still work.

If you're curious: here's the code that find a set of ports which IHP is going to use to start all it's services: https://github.com/digitallyinduced/ihp/blob/master/IHP/IDE/PortConfig.hs#L67 :)

Reply:
It seems to me like that the value that ToolServer writes to the IORef doesn't ever get read. I just removed it and starting a second project nicely picks a different set of ports.